extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/src/main/resources/org/apache/guacamole/auth/jdbc/user/PasswordRecordMapper.xml

```diff
@@ -66,12 +65,11 @@
         );
 
         DELETE FROM [guacamole_user_password_history]
-        WHERE password_history_id IN (
-            SELECT password_history_id
+        WHERE password_history_id NOT IN (
```

Comment:
Won't this now delete password history for all other users?

Reply:
I think I see what you're trying to do here now. Maybe needs an `AND user_id = #{record.userID,jdbcType=INTEGER}` at the level of the outer select?

Reply:
Ummm...yes.  Probably not the behavior we want.

Reply:
> I think I see what you're trying to do here now. Maybe needs an AND user_id = #{record.userID,jdbcType=INTEGER} at the level of the outer select?

Yep!  Commit coming right up...

Reply:
Should be fixed.